src/protocol.py

```diff
@@ -262,7 +262,9 @@ def assembleVersionMessage(remoteHost, remotePort, participatingStreams, server=
         payload += encodeHost('127.0.0.1')
         payload += pack('>H', 8444)
     else:
-        payload += encodeHost(remoteHost)
+        # use first 16 bytes if host data is longer
+        # for example in case of onion v3 service
+        payload += encodeHost(remoteHost)[:17]
```

Comment:
I think this is an off-by-one bug, it produces a string that is up to 17 bytes long (position 0 to 16). Or maybe I'm missing something.

Reply:
oops! There probably should be `[:16]`.